tests/functional/endpoint/test_endpoint_search.py

```diff
@@ -0,0 +1,139 @@
+        "location": "Automatic",
+        "max_concurrency": 4,
+        "max_parallelism": 8,
+        "my_effective_roles": [],
+        "myproxy_server": "myproxy.globusonline.org",
+        "name": collection_id,
+        "network_use": "normal",
+        "non_functional_endpoint_display_name": f"host of {display_name}",
+        "non_functional_endpoint_id": endpoint_id,
+        "owner_id": endpoint_id,
+        "owner_string": owner_string,
+        "preferred_concurrency": 2,
+        "preferred_parallelism": 4,
+        "public": True,
+        "shareable": True,
+        "tlsftp_server": f"tlsftp://{collection_fqdn}:443",
+        "username": username,
+    }
+    # there are so many null and False values in a typical item, it's worth packing them
+    # in strings and filling them in loops -- makes the document above more readable
+    null_keys = """
+        acl_max_expiration_period_mins authentication_assurance_timeout
+        authentication_policy_id contact_email contact_info department description
+        expire_time gcp_connected gcp_paused globus_connect_setup_key host_endpoint
+        host_endpoint_display_name host_endpoint_id host_path https_server info_link
+        keywords last_accessed_time local_user_info_available
+        mapped_collection_display_name mapped_collection_id myproxy_dn oauth_server
+        organization s3_url sharing_target_endpoint sharing_target_root_path
+        subscription_id user_message user_message_link
+    """.split()
+    false_keys = """
+        acl_available acl_editable activated disable_anonymous_writes disable_verify
+        force_encryption force_verify french_english_bilingual high_assurance in_use
+        is_globus_connect mfa_required non_functional requester_pays s3_owner_activated
+    """.split()
+    for k in null_keys:
+        data[k] = None
+    for k in false_keys:
+        data[k] = False
+    return data
+
+
+@pytest.fixture
+def singular_search_response():
+    collection_id = str(uuid.uuid4())
+    endpoint_id = str(uuid.uuid4())
+    return RegisteredResponse(
+        service="transfer",
+        path="/endpoint_search",
+        metadata={
+            "collection_id": collection_id,
+            "endpoint_id": endpoint_id,
+        },
+        json={
+            "DATA": [_make_mapped_collection_search_result(collection_id, endpoint_id)],
+            "DATA_TYPE": "endpoint_list",
+            "has_next_page": False,
+            "limit": 25,
+            "offset": 0,
+        },
+    )
+
+
+def test_search_shows_collection_id(run_line, singular_search_response):
+    singular_search_response.add()
+    meta = singular_search_response.metadata
+    collection_id = meta["collection_id"]
+    endpoint_id = meta["endpoint_id"]
+
+    result = run_line("globus endpoint search mytestquery")
+
+    lines = result.output.rstrip("\n").split("\n")
+    assert len(lines) == 3
+    assert endpoint_id not in lines[-1]
+    assert collection_id in lines[-1]
```

Comment:
Why is the final line important to check here? I can't read this and understand the significance of these tests.

Should `endpoint_id` _never_ appear in the output when a collection ID is known? If so, these tests are written with an assumption about conditions elsewhere in the code, and I recommend asserting that `endpoint_id` doesn't appear anywhere in `result.output`.

Reply:
Let me see if we have a nice matcher somewhere in CLI fixtures for this. The output is table-formatted and therefore has (with one result in the fixture data) the following format:
```
header field 1 | header field 2
---------------|---------------
value 1        | value 2
```

I should really assert, I think:
- that there are two lines at the start
- the first one is a header with the following fields: ... (fill this in)
- the second one is a separator line
- the rest of the lines are "data"
  - there is only one "data line"
  - that "data line" contains the collection_id
- the endpoint_id does not appear in the output

I can obviously do all that "from scratch", but I think we have some tools we made for this.

Reply:
We have tools for record-formatted output parsing, but not for tables (or at least, not in the places I looked). So I'll be doing this a bit manually, inline here for now.